envconfig.go

```diff
@@ -521,6 +521,16 @@ func lookup(key string, opts *options, l Lookuper) (string, bool, bool, error) {
 				if ok {
 					return s
 				}
+
+				// special case for the name of the env var itself, including
+				// its prefix
+				if i == "_" {
+					if pl, ok := l.(*prefixLookuper); ok {
+						return pl.prefix + key
+					}
+					return key
```

Comment:
This is exactly the same code as above.  Perhaps it should be extracted into a small utility function like:

```go
func actualKeyName(l Lookuper, key string) string {
	if pl, ok := l.(*prefixLookuper); ok {
		return pl.prefix + key
	}
	return key
}
```